### PR TITLE
Test on 3.12, drop support for 3.7

### LINKED ISSUE #95: Support python 3.12
Well, python 3.12 is out now and we should probably verify that taskgraph will work with 3.12.

### PR DESCRIPTION
**Pull request by phargogh**
No changes needed, but I did add a `pyproject.toml` to the project to update for the latest build standards.

Fixes #95

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -34,10 +34,10 @@
         'Operating System :: MacOS :: MacOS X',
         'Operating System :: Microsoft',
         'Operating System :: POSIX',
-        'Programming Language :: Python :: 3.7',
         'Programming Language :: Python :: 3.8',
         'Programming Language :: Python :: 3.9',
         'Programming Language :: Python :: 3.10',
         'Programming Language :: Python :: 3.11',
+        'Programming Language :: Python :: 3.11',
```
Comment (dcdenu4) [inline setup.py:41]:
```suggestion
        'Programming Language :: Python :: 3.12',

```

**Review by dcdenu4 — CHANGES_REQUESTED** [review]
Hey @phargogh just one minor thing I noticed in setup.py

**Comment by phargogh** [comment]
Whoops, thanks for catching that @dcdenu4 !

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    "Development Status :: 5 - Production/Stable",
+    "Natural Language :: English",
+    "Operating System :: MacOS :: MacOS X",
+    "Operating System :: Microsoft",
+    "Operating System :: POSIX",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.11",
```
Comment (dcdenu4) [inline pyproject.toml:23]: haha, I'm not sure what happened but this got flipped to 3.11! 
```suggestion
    "Programming Language :: Python :: 3.12",
```

**Review by dcdenu4 — CHANGES_REQUESTED** [review]
Hey @phargogh , something funky happened in pyproject.toml, 3.12 was swapped back to 3.11.

**Comment by phargogh** [comment]
Weird, let's try this again!

**Review by dcdenu4 — APPROVED** [review]
Hooray!

**Pull request merged**